packages/client/hmi-client/src/components/workflow/ops/model-comparison/tera-model-comparison.vue

```diff
@@ -461,25 +491,31 @@ function generateOverview(output: string) {
 }
 
 // Create a task to compare the models
-async function processCompareModels(modelIds: string[]) {
-	const taskRes = await compareModels(modelIds, props.node.workflowId, props.node.id);
+const processCompareModels = async () => {
+	isProcessingComparison.value = true;
+	const taskRes = await compareModels(modelIds.value, goalQuery.value, props.node.workflowId, props.node.id);
 	compareModelsTaskId = taskRes.id;
 	if (taskRes.status === TaskStatus.Success) {
 		generateOverview(taskRes.output);
 	}
-}
+	const state = cloneDeep(props.node.state);
+	state.hasRun = true;
+	emit('update-state', state);
+	isProcessingComparison.value = false;
+};
 
 // Listen for the task completion event
 useClientEvent(ClientEventType.TaskGollmCompareModel, (event: ClientEvent<TaskResponse>) => {
-	if (
-		!event.data ||
-		event.data.id !== compareModelsTaskId ||
-		!isEmpty(overview.value) ||
-		event.data.status !== TaskStatus.Success
-	) {
-		return;
+	if (!event.data || event.data.id !== compareModelsTaskId) {
+		console.log(event);
+	} else if ([TaskStatus.Queued, TaskStatus.Running, TaskStatus.Cancelling].includes(event.data.status)) {
```

Comment:
```suggestion
if ([TaskStatus.Queued, TaskStatus.Running, TaskStatus.Cancelling].includes(event.data.status)) {
```

Reply:
have to do early return if the task isn't the one on the model comparison node.